src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -402,7 +402,7 @@ public Task WriteWindowUpdateAsync(int streamId, int sizeIncrement)
                 _outgoingFrame.PrepareWindowUpdate(streamId, sizeIncrement);
                 WriteHeaderUnsynchronized();
                 var buffer = _outputWriter.GetSpan(4);
-                Bitshifter.WriteUInt31BigEndian(buffer, (uint)sizeIncrement);
+                BinaryPrimitives.WriteUInt32BigEndian(buffer, (uint)sizeIncrement);
```

Comment:
If you change this you need to ensure sizeIncrement is still only 31 bits so the reserve bit never gets set accidentally.

Reply:
Might as well still have WriteUInt31BigEndian helper method that checks the value fits in 31 bits and calls WriteUInt32BigEndian internally.

Reply:
Add a bool for preserveHighBit? We wouldn't preserve it, but it would clarify the intent.

Reply:
There are cases where this would need to be preserved, like the exclusive bit in priority frame. Though I'm not sure if we'll be implementing that. I'll add an overload to specify preservation.